### PR TITLE
Feature/complaint form review - part -1

### PR DESCRIPTION
**Pull request by rakeshkumar-verma**

### CODE REVIEW

File: mysql-schema/complaint-table.txt [hunk 1]
```diff
@@ -3,12 +3,14 @@ CREATE DATABASE `hclecomm` /*!40100 DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb
...
+  `id` int NOT NULL AUTO_INCREMENT,
+  `first_name` varchar(45) DEFAULT NULL,
+  `last_name` varchar(45) DEFAULT NULL,
+  `email` varchar(45) DEFAULT NULL,
+  `subject` varchar(45) DEFAULT NULL,
+  `complaint` varchar(500) DEFAULT NULL,
+  `status` varchar(45) DEFAULT NULL,
+  `closing_comment` varchar(500) DEFAULT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB AUTO_INCREMENT=24 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
```
Comment (rishabh-behl) [inline mysql-schema/complaint-table.txt]: @rakeshkumar-verma : why do we need below statement in query : ENGINE=InnoDB AUTO_INCREMENT=24 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;

**Review by rishabh-behl — COMMENTED** [review]
@rakeshkumar-verma : pl check comments also I can see 5-7 getting removed via this PR . Pl check if that is intended .

File: mysql-schema/complaint-table.txt (hunk 1 above)
Comment (rakeshkumar-verma) [reply mysql-schema/complaint-table.txt]: @rishabh-behl  I have export the table schema so its exported db engine, auto increment count.... Removed now.

File: ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml [hunk 2]
```diff
@@ -51,109 +53,185 @@
...
...ot;:&quot;BLOCK_STATEMENTS&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;BLOCK_STATEMENT&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;DISABLE_STATEMENT&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;AFCOMPONENT&quot;\,&quot;value&quot;:{&quot;id&quot;:&quot;guide.guideRootPanel.user-section&quot;\,&quot;displayName&quot;:&quot;User Section&quot;\,&quot;type&quot;:&quot;PANEL&quot;\,&quot;displayPath&quot;:&quot;FORM/Root Panel/User Section/&quot;\,&quot;name&quot;:&quot;user-section&quot;\,&quot;parent&quot;:&quot;guide.guideRootPanel&quot;}}]}}]}]}}]\,&quot;isValid&quot;:true\,&quot;enabled&quot;:true\,&quot;version&quot;:1\,&quot;script&quot;:&quot;/**\\n *  SHOW Support Section WHEN  (Hide Support Section is equal to )  ELSE Hide\\n *\\n * This is a machine-generated code for the rule.\\n * If you modify it in the code editor\, you will not be able to view and edit the rule in the visual editor.\\n */\\n\\nif (this.value == \\&quot;no\\&quot;) {\\n    guideBridge.resolveNode(\\&quot;user-section\\&quot;).enabled = false;\\n}\\n&quot;\,&quot;eventName&quot;:&quot;Value Commit&quot;\,&quot;ruleType&quot;:&quot;&quot;\,&quot;description&quot;:&quot;&quot;}]"
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml:171]: @rakeshkumar-verma : we have lot of characters here and maven install hanged for me in reading these charactes . can you check if all are required ?

File: ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml [hunk 3]
```diff
@@ -51,109 +53,185 @@
...
...name&quot;:&quot;hide_support_section&quot;\,&quot;parent&quot;:&quot;guide.guideRootPanel.user-section&quot;}}}\,{&quot;nodeName&quot;:&quot;OPERATOR&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;EQUALS_TO&quot;\,&quot;value&quot;:null}}\,{&quot;nodeName&quot;:&quot;EXPRESSION&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;STRING_LITERAL&quot;\,&quot;value&quot;:&quot;no&quot;}}]}}\,{&quot;nodeName&quot;:&quot;Else&quot;\,&quot;value&quot;:null}\,{&quot;nodeName&quot;:&quot;DONOTHING_OR_HIDE&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;Hide&quot;\,&quot;value&quot;:null}}]}}]\,&quot;isValid&quot;:true\,&quot;enabled&quot;:true\,&quot;version&quot;:1\,&quot;script&quot;:&quot;/**\\n *  SHOW Support Section WHEN  (Hide Support Section is equal to )  ELSE Hide\\n *\\n * This is a machine-generated code for the rule.\\n * If you modify it in the code editor\, you will not be able to view and edit the rule in the visual editor.\\n */\\n\\nif (hide_support_section.value == \\&quot;no\\&quot;) {\\n    true;\\n } else {\\n    false;\\n}\\n&quot;\,&quot;eventName&quot;:&quot;Visibility&quot;\,&quot;ruleType&quot;:&quot;&quot;\,&quot;description&quot;:&quot;&quot;}]"
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml:231]: @rakeshkumar-verma : we have lot of characters here and maven install hanged for me in reading these charactes . can you check if all are required ?

File: ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml [hunk 4]
```diff
@@ -174,6 +253,53 @@
...
...:&quot;EVENT_CONDITION&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;EVENT_AND_COMPARISON&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;COMPONENT&quot;\,&quot;value&quot;:{&quot;id&quot;:&quot;XX&quot;\,&quot;type&quot;:&quot;BUTTON&quot;\,&quot;name&quot;:&quot;XX&quot;}}\,{&quot;nodeName&quot;:&quot;EVENT_AND_COMPARISON_OPERATOR&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;is clicked&quot;\,&quot;value&quot;:null}}\,{&quot;nodeName&quot;:&quot;PRIMITIVE_EXPRESSION&quot;\,&quot;choice&quot;:null}]}\,&quot;nested&quot;:false}\,{&quot;nodeName&quot;:&quot;Then&quot;\,&quot;value&quot;:null}\,{&quot;nodeName&quot;:&quot;BLOCK_STATEMENTS&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;BLOCK_STATEMENT&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;SUBMIT_FORM&quot;\,&quot;items&quot;:[]}}]}]}}]\,&quot;isValid&quot;:true\,&quot;enabled&quot;:true\,&quot;version&quot;:1\,&quot;script&quot;:&quot;/**\\n\\n * This is a machine-generated code for the rule.\\n * If you modify it in the code editor\, you will not be able to view and edit the rule in the visual editor.\\n */\\n\\nguideBridge.submit();\\n\\n&quot;\,&quot;eventName&quot;:&quot;Click&quot;}]"
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml:279]: @rakeshkumar-verma : we have lot of characters here and maven install hanged for me in reading these charactes . can you check if all are required ?

File: ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml [hunk 5]
```diff
@@ -174,6 +253,53 @@
...
...t;:&quot;EVENT_CONDITION&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;EVENT_AND_COMPARISON&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;COMPONENT&quot;\,&quot;value&quot;:{&quot;id&quot;:&quot;XX&quot;\,&quot;type&quot;:&quot;BUTTON&quot;\,&quot;name&quot;:&quot;XX&quot;}}\,{&quot;nodeName&quot;:&quot;EVENT_AND_COMPARISON_OPERATOR&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;is clicked&quot;\,&quot;value&quot;:null}}\,{&quot;nodeName&quot;:&quot;PRIMITIVE_EXPRESSION&quot;\,&quot;choice&quot;:null}]}\,&quot;nested&quot;:false}\,{&quot;nodeName&quot;:&quot;Then&quot;\,&quot;value&quot;:null}\,{&quot;nodeName&quot;:&quot;BLOCK_STATEMENTS&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;BLOCK_STATEMENT&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;RESET_FORM&quot;\,&quot;items&quot;:[]}}]}]}}]\,&quot;isValid&quot;:true\,&quot;enabled&quot;:true\,&quot;version&quot;:1\,&quot;script&quot;:&quot;/**\\n\\n * This is a machine-generated code for the rule.\\n * If you modify it in the code editor\, you will not be able to view and edit the rule in the visual editor.\\n */\\n\\nguideBridge.reset();\\n\\n&quot;\,&quot;eventName&quot;:&quot;Click&quot;}]"
```
Comment (rishabh-behl) [inline ui.content/src/main/content/jcr_root/content/forms/af/hclecomm/user-complaint-form/.content.xml:295]: @rakeshkumar-verma : we have lot of characters here and maven install hanged for me in reading these charactes . can you check if all are required ?

**Review by rishabh-behl — APPROVED** [review]
lgtm

**Pull request merged**